templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -0,0 +1,313 @@
+            exit 1
+          fi
+          phys_port_opts+=" ovs-port.bond-mode ${ovs_interfaces__bridge_port__link_aggregation_mode}"
+        else
+          # Check for LACP, valid settings are "active", "off", "passive"
+          VALID_LACP=("active" "off" "passive")
+          if [ ! -z ${ovs_interfaces__bridge_port__link_aggregation_lacp=:-} ]; then
+            if [[ ! " ${VALID_LACP[@]} " =~ " ${ovs_interfaces__bridge_port__link_aggregation_lacp} " ]]; then
+              echo "ERROR: Invalid value specified for LACP: ${ovs_interfaces__bridge_port__link_aggregation_lacp}"
+              exit 1
+            fi
+            phys_port_opts+=" ovs-port.lacp ${ovs_interfaces__bridge_port__link_aggregation_lacp}"
+          else
+            # Default to active/backup mode
+            phys_port_opts+=" ovs-port.bond-mode active-backup"
+          fi
+        fi
+      else
+        # check if port is specified if no bonding is used
+        if [ ! -z ${ovs_interfaces__bridge_port__name:-} ]; then
+          for port in ${ovs_interfaces__bridge_port__name[@]}; do
+            if [ "$port" != "br-ex" ]; then
+              config_iface=$port
+            fi
+          done
+        fi
+      fi
+
+      # Override detected interface with one in config file if present.
+      # This will only be used for trying to detect MTU or pick a MAC address if not provided in the config file
+      if [ -z "$config_iface" ]; then
+        echo "WARN: No bonding or physical interface specified in config file!"
+      else
+        echo "Interface detected via config file: ${config_iface}"
+        iface=$config_iface
+      fi
+      echo "Bonding and VLAN options found via config file: ${phys_port_opts}"
+    fi
+
+    if [ -z "$iface" ]; then
+      echo "ERROR: Unable to find default gateway interface and no configuration provided"
+      exit 1
+    fi
+
+    # find the MAC from OVS config or the default interface to use for OVS internal port
+    # this prevents us from getting a different DHCP lease and dropping connection
+    if ! iface_mac=${ovs_interfaces__mac_address[0]:-$(<"/sys/class/net/${iface}/address")}; then
+      echo "Unable to determine default interface MAC"
+      exit 1
+    fi
+
+    echo "MAC address found for iface: ${iface}: ${iface_mac}"
+
+    # find MTU from config or on original iface
+    iface_mtu=${ovs_interfaces__mtu[0]:-$(ip -j link show "$iface" | jq -r '.[0].mtu')}
+    if [ -z "$iface_mtu" ]; then
+      echo "Unable to determine default interface MTU, defaulting to 1500"
+      iface_mtu=1500
+    else
+      echo "MTU found for iface: ${iface}: ${iface_mtu}"
+    fi
+
+    # create bridge
+    if ! nmcli connection show br-ex &> /dev/null; then
+      nmcli c add type ovs-bridge conn.interface br-ex con-name br-ex 802-3-ethernet.mtu ${iface_mtu} 802-3-ethernet.cloned-mac-address ${iface_mac}
+    fi
+
+    old_conns=()
+    if "$BONDING"; then
+      iface=bond0
```

Comment:
why hard-code to bond0? should this be `${bonded_ifaces[0]}` or whatever is the primary bond interface?

Reply:
I should parse this from the config file. It will be under the ovs_interfaces__bridge_port__name array.